roles/splunk/tasks/remove_splunk.yml

```diff
@@ -11,4 +11,121 @@
     state: absent
   become: true
 
-# Todo: Remove user, group, THP/ulimit files, check for cron jobs/scripts and remove those if present
+- name: Block for non-root splunk user removal
+  block:
+    - name: Remove nix splunk user
+      ansible.builtin.user:
+        name: "{{ splunk_nix_user }}"
+        groups: "{{ splunk_nix_group }}"
+        state: absent
+        local: "{{ local_os_user }}"
+      become: true
+
+    - name: Remove nix splunk group
+      ansible.builtin.group:
+        name: "{{ splunk_nix_group }}"
+        state: absent
+        local: "{{ local_os_group }}"
+      become: true
+  when: splunk_nix_user != 'root'
+
+- name: Remove Disable-THP/Ulimit files
```

Comment:
There is also a `tuned` profile to disable THP, but I think you can leave that in there for now, because we don't really know what profile it was using prior to the change. Unless we record somewhere what he original profile was. But that's a whole other thing that I wanted to do (adding a comment in the new file what the original was), and is probably outside the scope of this PR.

Reply:
Agreed, the tuned profile switch is a great option for people who aren't already using tuned, and I'm sure they can change it accordingly in their removal file.